src/text-fragment-utils.js

```diff
@@ -820,3 +820,24 @@ if (typeof goog !== 'undefined') {
   goog.declareModuleId('googleChromeLabs.textFragmentPolyfill.textFragmentUtils');
   // clang-format on
 }
+
+/**
+ * Seaches ::target-text background color in the document
+ *
+ * @return {string} - background color from ::target-text
+ */
+export const getTargetTextBackgroundColor = () => {
+  const styles = document.getElementsByTagName("style");
+  if (!styles) return null;
+
+  for (const style of styles) {
+    const cssRules = style.innerHTML;
+    const targetTextRules = cssRules.match(/::target-text\s*{\s*((.|\n)*?)\s*}/g);
+    if (!targetTextRules) return null;
+
+    const backgroundColor = targetTextRules[0].match(/background-color\s*:\s*(.*?)\s*;/);
+    if (!backgroundColor) return null;
+    
+    return backgroundColor[1];
```

Comment:
It seemed to work fine with the few test cases used but we could add some regex validation on the result. i.e. `Hex`, `RGB`, or `color names`.. validations.

Reply:
Sorry I don't think my comment was super clear, what I mean is are we sure that `backgroundColor[1]` will always always exist or should we do something like
```
return backgroundColor.length >= 2 ? backgroundColor[1] : null;
```

Or I guess maybe that's not strictly necessary since it would just return `undefined` anyway?

Reply:
Oh yes out of range, sorry I misread your comment. You're right, I believe it would just return `undefined`, but I am ok with checking the size of array before returning.

Reply:
Ok, let's be explicit about it and add the size check.